### PR TITLE
CLN: deprecate the pandas.tseries.plotting.tsplot function (GH18627)

### LINKED ISSUE #18627: CLN: deprecate/remove the pandas.tseries.plotting.tsplot function
xref https://github.com/pandas-dev/pandas/pull/17814

As far as I can see, the `tsplot` function (https://github.com/pandas-dev/pandas/blob/master/pandas/plotting/_timeseries.py#L23) is not used internally, and it is not exposed in the `pandas.plotting` namespace (it *is* still exposed in `pandas.tseries.plotting`). 
And I also don't think it is actually a useful function for users? (they would just do `ts.plot()`, and here you have to awkwardly provide an actual plotting function)

### PR DESCRIPTION
**Pull request by GGordonGordon**
- [X] closes #18627 
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -678,6 +678,8 @@ Deprecations
 - The ``reduce`` parameter of ``.apply()`` is deprecated in favor of ``result_type='reduce'`` (:issue:`18577`)
 - The ``order`` parameter of :func:`factorize` is deprecated and will be removed in a future release (:issue:`19727`)
 
+- ``pandas.tseries.plotting.tsplot`` is deprecated. Use :func:`TimeSeries.plot` instead (:issue:`18627`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.23.0.txt]: TimeSeries -> Series

File: pandas/plotting/_timeseries.py [hunk 2]
```diff
@@ -35,7 +36,14 @@ def tsplot(series, plotf, ax=None, **kwargs):
     _____
     Supports same kwargs as Axes.plot
 
+
+    .. deprecated:: 0.23.0
+       Use TimeSeries.plot() instead
     """
+    warnings.warn("'tsplot' is deprecated and will be removed in a "
+                  "future version. Please use TimeSeries.plot() instead.",
```
Comment (jorisvandenbossche) [inline pandas/plotting/_timeseries.py]: same here

**Review by jorisvandenbossche — COMMENTED** [review]
Minor comment, for the rest looks good!

(seeing how it has to be used (with passing a plot function), I very much wonder if actually somebody is using this ..)

**Comment by GGordonGordon** [comment]
@jorisvandenbossche - I've gone ahead and made the rewording changes (TimeSeries -> Series) you've requested and squashed everything together for cleanliness.

**Comment by jorisvandenbossche** [comment]
Thanks a lot!

> and squashed everything together for cleanliness.

For future reference, you don't need to bother doing that. We squash everything on merging the PR anyway, and keeping the multiple commits makes it easier what was updated after a review.

**Pull request merged**